kymatio/scattering2d/filter_bank.py

```diff
@@ -171,9 +171,9 @@ def gabor_2d(M, N, sigma, theta, xi, slant=1.0, offset=0):
         morlet_fft : ndarray
             numpy array of size (M, N)
     """
-    gab = np.zeros((M, N), np.complex64)
```

Comment:
I removed any type hard-coding

Reply:
I think it would be better to hard-code, actually. That way it's clear from the docs what the default dtype is.

Reply:
the default type of np is user dependant na? It can be changed, isn't it?

Reply:
It is? How? Still, that's quite a roundabout way of doing it. Here's what I suggest: the default precision is double precision (`np.float64`/`np.complex128`). We then reimplement the same functions that we have in Torch to change the precision: `float()` (to get single precision) and `double()` (to get double precision). This way, the precision is user-defined, but in an easy-to-access way. A slight problem here would be that `scattering.float().double()` won't necessary return the original double-precision filters due to truncation error, but from what I understand, the Torch frontend has the same behavior, so maybe we can just leave it there for now.

Reply:
I'd be pros the opposite: float32 and complex64 as the default because this is usually what is expected. I agree with your concern. Or maybe simply should we fix to work only with float32?

Reply:
Ok. How about this? Let's leave it at the NP default (as you have), *but* we relax the CDGMM function so that it doesn't require the same underlying type. We won't have the same interface as the `torch` backends, but we can address that later if we want. (Or not. In NP, you can multiply two arrays of different dtypes and get a sensible result. In Torch, you can't. It might make sense if our frontends mirrored that difference.)

Reply:
Sorry, let me amend that. We leave it at *double precision*, but relax the constraints on the CDGMM.

Reply:
Gah. I mean *single precision*. Like we had it before. We just don't want it to error if people put in double-precision arrays.